packages/noya-designsystem/src/components/GridView.tsx

```diff
@@ -64,37 +64,74 @@ const ItemDescription = styled.span(({ theme }) => ({
   textOverflow: 'ellipsis',
 }));
 
+type MenuItemType = 'delete' | 'copy';
+
 interface ItemProps {
-  children?: ReactNode;
-  title: string;
+  id?: string;
```

Comment:
Ideally, the `MenuItemType` should be generic here. If `ItemProps` and `GridItem` are generic (`<MenuItemType extends string>`) then they can work with other kinds of menu items. We use generics in `ListView.Row`:

```ts
export interface ListViewRowProps<MenuItemType extends string = string> { ... }

function ListViewRow<MenuItemType extends string>(...) { ... }
```

In this case it will be:

```ts
export interface ItemProps<MenuItemType extends string = string> { ... }

function GridItem<MenuItemType extends string>(...) { ... }
```

You may also have to render it as `<GridItem<MenuItemType> ... />` in the other grids